hs/app/reach/Main.hs

```diff
@@ -1996,66 +2113,227 @@ versionCompare2 = command "version-compare2" $ info f mempty
       let reaches = query t . Right . ("reachsh/" <>) <$> rights imagesAll
       let others = lefts imagesAll <&> \o -> query (imageThirdPartyTagRaw o) $ Left o
       let both = reaches <> others
-      let lefty = either (T.toLower . packs) id
 
-      let uImgs = [a | DAQUnknownImg a _ <- both]
-      let uTags = [(x, y) | DAQUnknownTag x y <- both]
-      let synced = [(x, y, z) | DAQSync x y z <- both]
-      let newDAs = [(x, y, z) | DAQNewDigestAvailable x y z <- both]
-      let newTags = [(x, y, z) | DAQNewTags x y z <- both]
-      let newCAs = [(x, y, z) | DAQNewConnectorAvailable x y z <- both]
+      let uImgs = [Image' a | DAQUnknownImg a _ <- both]
+      let uTags = [(Image' x, y) | DAQUnknownTag x y <- both]
 
-      liftIO $ do
-        if length uImgs > 0
-          then do
-            forM_ uImgs $ \a -> T.putStrLn $ "Unknown image: " <> lefty a <> "."
-            T.putStrLn $ "Please open an issue at " <> uriIssues <> " including the failures listed above."
-            exitWith $ ExitFailure 1
-          else
-            if length uTags > 0
-              then do
-                forM_ uTags $ \(x, y) -> T.putStrLn $ "Unknown tag: " <> y <> " for image: " <> lefty x <> "."
-                exitWith $ ExitFailure 1
-              else
-                if length (synced <> newCAs) >= length both
-                  then do
-                    putStrLn "Reach is up-to-date."
-                    exitWith $ ExitFailure 0
-                  else do
-                    let s = T.take 8 . T.drop 7
-                    let m =
-                          maybe 0 id . maximumMay $
-                            (\(x, _, _) -> T.length $ lefty x)
-                              <$> newDAs <> newTags <> newCAs <> synced
+      let vc@VersionCompare{..} = VersionCompare
+            [VersionCompareIDTs (Image' x) y z | DAQSync x y z <- both]
+            [VersionCompareIDTs (Image' x) y z | DAQNewDigestAvailable x y z <- both]
+            [VersionCompareIDTs (Image' x) y z | DAQNewTags x y z <- both]
+            [VersionCompareIDTs (Image' x) y z | DAQNewConnectorAvailable x y z <- both]
 
-                    let p x = lefty x <> T.replicate (m - T.length (lefty x)) " "
-                    let n a = when (any ((> 0) . length) a) $ putStrLn ""
+      if length uImgs > 0
```

Comment:
Eww, I really hate the way `if then else` looks in Haskell. The formatting always messes me up to the point that I always use `case`. I also often do `case (x, y, z)` so I can get them all at the same indentation too

Reply:
I really don't like how this looks either but `hs-format` thinks it's better somehow. I thought this was way easier to read:

https://github.com/reach-sh/reach-lang/commit/2518e01b7c0094340d00bbcc2ff22973dd80afd9#diff-ceefed8a152f4a53d8a11bd2bc7dabb608c98d48cce9c3d0f4484a0451b6ab26L1838-L1881

```haskell
 liftIO $ do
    if length uImgs > 0 then do
      forM_ uImgs $ \a -> T.putStrLn $ "Unknown image: " <> lefty a <> "."
      T.putStrLn $ "Please open an issue at " <> uriIssues <> " including the failures listed above."
      exitWith $ ExitFailure 1

    else if length uTags > 0 then do
      forM_ uTags $ \(x, y) -> T.putStrLn $ "Unknown tag: " <> y <> " for image: " <> lefty x <> "."
      exitWith $ ExitFailure 1

    else if length (synced <> newCAs) >= length both then do
      putStrLn "Reach is up-to-date."
      exitWith $ ExitFailure 0

    else do
      ...
```

I'll try to tidy it up with a `case` but I think that might look really crowded for this block of code.